### DISCUSSION TITLE
Is there a way to get the two YAML files only the keys that appear only twice? For example:

### DISCUSSION THREAD

**Discussion by omrv** [mikefarah/yq#1652] [Q&A]
file1.yaml

name: ABB
host: CCC
age: 120
TSG: TTT

file2.yaml

host: CCC
age: 120
TST: TTT

output:

host: CCC
age: 120

**Comment by omrv** [comment]
c

**Comment by mikefarah** [comment]
Yep - here's how it works at a high level:

- We can use the 'key' operator to return an array of keys of a particular map
- We can subtract two arrays to find the difference between them
- Therefore, we can find the keys in both maps by removing the keys in file1 that are not in file2

Explanation:
- Lets get the keys that from file1 that are not in file 2, the keys in file 1 is just `keys` (because we are running against that file). The keys in file 2 are `(load("file2.yaml") | keys`. The difference between them is then `(keys - (load("file2.yaml") | keys))`
- That will return us the keys that are in file1, but not file 2. We want to remove those, so we do `keys - (keys - (load("file2.yaml") | keys))`
- Finally, the output you had has both keys and values from file1, we can use pick to do that

```
yq 'pick(keys - (keys - (load("file2.yaml") | keys)))' file1.yaml
```

**Comment by omrv** [comment]
Thanks!

Is there a way to get all the unique keys from both files?
